code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm

```diff
@@ -408,14 +408,14 @@
 	location.ClearWet()
 	if(location.air)
 		var/datum/gas_mixture/air = location.air
-		air.temperature = 293.15
+		air.temperature = T20C
 		for(var/obj/effect/hotspot/fire in location)
 			qdel(fire)
 
 		var/list/gases = air.gases
 		for(var/gas_type in gases)
 			switch(gas_type)
-				if(/datum/gas/oxygen, /datum/gas/nitrogen)
+				if(/datum/gas/oxygen, /datum/gas/nitrogen, /datum/gas/halon)
 					continue
```

Comment:
while youre touching this, this part feels really weird and i have no idea why its a switch type. can you just make it
if(gas_type in list(/datum/oxygen, etc))

Reply:
Yeah.